block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -1034,6 +1082,121 @@ object BlockDagFileStorage {
             }
     } yield LMDBStore[F](env, dbi)
 
+  private def readBlockHashesByDeploy[F[_]: Sync](
+      randomAccessIO: RandomAccessIO[F]
+  ): F[List[(DeployId, BlockHash)]] = {
+    def readRec(
+        result: List[(DeployId, BlockHash)]
+    ): F[List[(DeployId, BlockHash)]] =
+      for {
+        deployIdSizeOpt <- randomAccessIO.readInt
+        result <- deployIdSizeOpt match {
+                   case Some(deployIdSize) =>
+                     val blockHashBytes = Array.ofDim[Byte](BlockHash.Length)
+                     val deployIdBytes  = Array.ofDim[Byte](deployIdSize)
+                     for {
+                       readDeployId  <- randomAccessIO.readFully(deployIdBytes)
+                       readBlockHash <- randomAccessIO.readFully(blockHashBytes)
+                       result <- (readDeployId, readBlockHash) match {
+                                  case (Some(()), Some(())) =>
+                                    readRec(
+                                      (ByteString.copyFrom(deployIdBytes) -> ByteString
+                                        .copyFrom(blockHashBytes)) :: result
+                                    )
+                                  case _ =>
+                                    Sync[F].raiseError[List[(DeployId, BlockHash)]](
+                                      BlockHashesByDeployLogIsCorrupted
+                                    )
+                                }
+                     } yield result
```

Comment:
Looks as if this whole method (and all its siblings) should be replaced by scodec, or a similar tool (In a separate JIRA/PR ofc). WDYT?